src/server/game/AI/PlayerAI/PlayerAI.cpp

```diff
@@ -192,10 +1277,48 @@ void SimpleCharmedPlayerAI::UpdateAI(const uint32 /*diff*/)
                 return;
 
             if (IsRangedAttacker())
-                AttackStartCaster(target, 28.0f);
+            {
+                if ((_chaseCloser = !me->IsWithinLOSInMap(target)))
+                    AttackStart(target);
+                else
+                    AttackStartCaster(target, CASTER_CHASE_DISTANCE);
+            }
             else
                 AttackStart(target);
         }
+
+        if (me->IsStopped() && !me->HasUnitState(UNIT_STATE_CANNOT_TURN))
+        {
+            float targetAngle = me->GetAngle(target);
+            if (fabs(me->GetOrientation() - targetAngle) > 0.4f)
+                me->SetFacingTo(targetAngle);
+        }
+
+        if (_castCheckTimer <= diff)
+        {
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                _castCheckTimer = 0;
+            else
+            {
+                if (IsRangedAttacker())
+                { // chase to zero if the target isn't in line of sight
+                    bool inLOS = me->IsWithinLOSInMap(target);
+                    if (_chaseCloser != !inLOS)
+                    {
+                        if ((_chaseCloser = !inLOS))
+                            AttackStart(target);
```

Comment:
AttackStart does attack capability checks (am I not feared/cced/other). Wouldn't want sheeped players suddenly starting to movechase.